src/v/raft/consensus.cc

```diff
@@ -2091,6 +2113,12 @@ ss::future<> consensus::write_snapshot(write_snapshot_cfg cfg) {
         co_return;
     }
 
+    if (ntp() == model::controller_ntp) {
+        // Skip prefix truncating the controller partition to preserve the
+        // controller log (in case we need to turn off snapshots).
+        co_return;
+    }
+
```

Comment:
Can we make this configurable from outside ? i.e. add an option to contstructor ?

Reply:
done